test/execution/execution-afm.test.js

```diff
@@ -0,0 +1,255 @@
+// Copyright (C) 2007-2014, GoodData(R) Corporation. All rights reserved.
+import fetchMock from '../utils/fetch-mock';
+import executionAfm, { nextPageOffset, mergePageData } from '../../src/execution/execute-afm';
+
+describe('nextPageOffset', () => {
+    it('should work for 1 dimension', () => {
+        expect(nextPageOffset({ offset: [0], total: [501] })).toEqual([500]);
+        expect(nextPageOffset({ offset: [500], total: [501] })).toEqual(false);
+        expect(nextPageOffset({ offset: [0], total: [1] })).toEqual(false);
+    });
+    it('should work for 2 dimensions', () => {
+        expect(nextPageOffset({ offset: [0, 0], total: [501, 501] })).toEqual([0, 500]);
+        expect(nextPageOffset({ offset: [0, 500], total: [501, 501] })).toEqual([500, 0]);
+        expect(nextPageOffset({ offset: [500, 0], total: [501, 501] })).toEqual([500, 500]);
+        expect(nextPageOffset({ offset: [500, 500], total: [501, 501] })).toEqual(false);
+    });
+    it('should work for 3 dimensions', () => {
+        expect(nextPageOffset({ offset: [0, 0, 0], total: [501, 501, 501] })).toEqual([0, 0, 500]);
+        expect(nextPageOffset({ offset: [0, 0, 0], total: [501, 501, 501] })).toEqual([0, 0, 500]);
```

Comment:
Is it exactly the same thing as the line above?

Reply:
oh yes, thanks for pointing out. Fixing.